sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultKeyStore.java

```diff
@@ -98,16 +101,40 @@ public KeyVaultKeyStore() {
         String clientSecret = System.getProperty("azure.keyvault.client-secret");
         String managedIdentity = System.getProperty("azure.keyvault.managed-identity");
         if (clientId != null) {
-            keyVaultClient = new KeyVaultClient(keyVaultUri, tenantId, clientId, clientSecret);
+            setKeyVaultClient(new KeyVaultClient(keyVaultUri, tenantId, clientId, clientSecret));
         } else {
-            keyVaultClient = new KeyVaultClient(keyVaultUri, managedIdentity);
+            setKeyVaultClient(new KeyVaultClient(keyVaultUri, managedIdentity));
         }
     }
 
+    private static void setKeyVaultClient(KeyVaultClient keyVaultClient) {
+        KeyVaultKeyStore.keyVaultClient = keyVaultClient;
+    }
+
+    private static void setAliases(List<String> aliases) {
+        KeyVaultKeyStore.aliases = aliases;
+    }
+
+    /**
+     * refresh certificate and KeyVaultTrustManager trustManager
+     */
+    public static void refreshCertificate() {
+        Optional.ofNullable(aliases)
+                .orElse(Collections.emptyList())
+                .forEach(alias -> {
+                    Key key = keyVaultClient.getKey(alias, null);
+                    Certificate certificate = keyVaultClient.getCertificate(alias);
+                    if (!Objects.isNull(key) && !Objects.isNull(certificate)) {
+                        CERTIFICATE_KEYS.put(alias, key);
```

Comment:
should this map be cleaned up before put?

Reply:
Since the side-load certificates will be added when the certificate is loaded. If it is cleaned and reloaded, the side-load local certificate may be deleted.

Reply:
Maybe we should store side-load certificates separately from those on the portal.
In this way, we do not modify the side-load certificates every time we refresh, and we can also set aliases to not static cause we get all aliases from portal.